client/cart/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import { recordUserEvent } from 'tracks';
+import { getConfig } from 'wcpay/utils/checkout';
+import WooPayDirectCheckout from 'wcpay/checkout/woopay/direct-checkout/woopay-direct-checkout';
+
+const addProceedToCheckoutTracking = () => {
+	Object.values( WooPayDirectCheckout.redirectElements ).forEach(
+		( className ) => {
+			const proceedButton = document.querySelector( className );
+
+			if ( ! proceedButton ) {
+				return;
+			}
+
+			const isWooPayEnabled = WooPayDirectCheckout.isWooPayEnabled();
+			const isWoopayDirectCheckoutEnabled = Boolean(
+				getConfig( 'isWooPayDirectCheckoutEnabled' )
+			);
+			proceedButton.addEventListener( 'click', () => {
```

Comment:
Great catch, @lovo-h! I've added (https://github.com/Automattic/woocommerce-payments/pull/8247/commits/4e490cb232a7a292cf8632e839aad56c77687e3c) an observer to watch the re-rendering of that section and re-attach the event listener as needed.

Can you please re-test this? My suggestion for testing it would be doing sequential updates in the cart and checking if the event is still being triggered correctly. For instance:

- 1st attempt: No updates in the cart.
- 2nd attempt: 1 update in the cart.
- 3rd attempt: 2 updates in the cart.
- 4th attempt: 3 updates in the cart.

Ifa single event is fired in all scenarios, then it's working as expected.